qemu/tests/readonly_disk.py

```diff
@@ -32,16 +31,33 @@ def run(test, params, env):
         "TEST STEPS 2: Format the disk and copy file to it", logging.info)
     os_type = params["os_type"]
     copy_cmd = params.get("copy_cmd", "copy %s %s")
-    disk_idx = params.get("disk_index", 1)
-    fs_type = params.get("fstype", "ntfs")
-    drive_letter = params.get("drive_letter", "I:")
-    disk_size = params.get("partition_size_data", "200M")
+    fstype = params.get("fstype", "ntfs")
+    data_image_size = params.get("image_size_data", "1G")
+    data_image_num = int(params.get("data_image_num",
+                                    len(params.objects("images")) - 1))
+    err_msg = "Set disks num: %d" % data_image_num
+    err_msg += ", Get disks num in guest: %d"
+    error_context.context("Get windows disk index that to "
+                          "be formatted", logging.info)
+    disks = utils_disk.get_windows_disks_index(session, data_image_size)
+    if len(disks) < data_image_num:
+        test.fail("Fail to list all the volumes"
+                  ", %s" % err_msg % len(disks))
     src_file = utils_misc.set_winutils_letter(
         session, params["src_file"], label="WIN_UTILS")
-    utils_misc.format_guest_disk(session, disk_idx, drive_letter,
-                                 disk_size, fs_type, os_type)
-    dst_file = params["dst_file"]
-    session.cmd(copy_cmd % (src_file, dst_file))
+    error_context.context("Clear readonly for all disks and online "
+                          "them in guest.", logging.info)
+    if not utils_disk.update_windows_disk_attributes(session, disks):
+        test.fail("Failed to update windows disk attributes.")
+    for disk in disks:
+        error_context.context("Format disk in guest: '%s'" % disk,
```

Comment:
it's better change to error_context.context("Format disk %s in guest" % disk, logging.info) .
currently code will print info as " Format disk in guest: '1' ", user will confuse the meaning for end number '1'.

Reply:
disk index mentioned in line 40, error_context.context(\"Get windows disk index that to be formatted\", logging.info)

Reply:
Althrough disk index mentioned in line 40, but like this comment "01:43:41 INFO | Context: Format disk in guest: '1'",  I thinks it's so confusing user if not check the code source~